src/services/expenseService.ts

```diff
@@ -1,119 +1,141 @@
 import { Expense } from "../domain/expense";
 
-import moment from 'moment';
-
 export class ExpenseService {
-    private expenseListDB: Array<Expense>;
-
-    constructor(expenseListDB: Array<Expense>) {
-        this.expenseListDB = expenseListDB;
-    }
-
-    public getExpenses(): Array<Expense> {
-        return this.expenseListDB;
-    }
 
-    public getExpensesByUserId(userId: number): Array<Expense> {
-        let resultList = this.expenseListDB.filter((expense) => expense.userId === userId);
-        return resultList;
+    constructor() {
     }
 
-    public getExpenseByUserIdAndExpenseId(userId: number, expenseId: number): Expense {
-        let resultList = this.expenseListDB.filter((expense) => expense.userId === userId && expense.id === expenseId);
-        return resultList[0];
-    }
-
-    public addExpense(userId: number, name: string, amount: bigint, date: Date, categoryId: number ): boolean {
-            let newExpenseId;
-            let expenseList = this.expenseListDB;
-            if (expenseList.length == 0) {
-                newExpenseId = 1;
+    public async getExpenses(userId: number): Promise<Expense[]> {
+        try {
+            const response = await window.fetch("http://localhost:3001/api/v1/users/" + userId.toString() + "/expenses", {
+                method: "GET",
+                headers: {
+                    Accept: "application/json"
+                }
+            });
+    
+            if (!response.ok) {
+                throw new Error(`Error! status: ${response.status}`);
+            }
+    
+            const result = (await response.json());
+    
+            // console.log("result is: ", JSON.stringify(result, null, 4));
+    
+            return <Expense[]>JSON.parse(JSON.stringify(result, null, 4));
+    
+        } catch (error) {
+            if (error instanceof Error) {
+                console.log('error message: ', error.message);
             } else {
-                let lastExpense = expenseList[expenseList.length - 1];
-                newExpenseId = lastExpense.id + 1;
+                console.log('unexpected error: ', error);
             }
-            let newExpense = new Expense(userId, newExpenseId, name, amount, date, categoryId);
-            console.log("newExpense", newExpense);
-            this.expenseListDB.push(newExpense);
-            return true;
-    }
-
-    public editExpense(userId: number, id: number, editedName: string, editedAmount: bigint, editedDate: Date, editedCategoryId: number): boolean {
-        let expense = this.getExpenseByUserIdAndExpenseId(userId, id);
-        let index = this.getExpenses().indexOf(expense);
-        let editedExpense = new Expense(userId, id, editedName, editedAmount, editedDate, editedCategoryId);
-        this.expenseListDB[index] = editedExpense;
-        console.log("editedExpense", editedExpense);
-        return true;
+            return null as any;
+        }
```

Comment:
if ve else blokları aynı işi yapıyor gibi duruyor. Ayrıca null dönmek uygulamada istenmeyen sonuçlara yol açabilir. 
`throw 'Parameter is not a number!';` gibi bir kullanım ihtiyacını görür mü ?